### DISCUSSION TITLE
How do I unmap tab and shift tab for snippets completion?

### DISCUSSION THREAD

**Discussion by sainad2222** [LazyVim/LazyVim#4307] [Q&A] [answered]
I had currently the following configs in plugins folder `coding.lua`

```
return {
  "nvim-cmp",
  keys = {
    {
      "<Tab>",
      mode = { "i", "s" },
      false,
    },
    {
      "<S-Tab>",
      mode = { "i", "s" },
      false,
    },
  },
}

```

```
return {
  "nvim-cmp",
  keys = {
    {
      "<Tab>",
      "<Tab>",
      mode = { "i", "s" },
    },
    {
      "<S-Tab>",
      "<S-Tab>",
      mode = { "i", "s" },
    },
  },
}

```

But this is not unmapping Tab and Shift tab. TIA

**Comment by dpetka2001** [comment]
https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps

**Reply by sainad2222** [reply]
Hey @dpetka2001, Thanks for the reply. But I referred to the docs you attached already and added that config as reference too. It mentioned to give lhs and keep false after thay to disable but it was not working for me

**Reply by dpetka2001** [reply]
> caution
Make sure to use the exact same mode as the keymap you want to disable. You don't have to specify a mode for normal mode keymaps.

You have to also define the modes you want the keymaps disabled. Both `<Tab>` and `<S-tab>` are defined for `mode = { "i", "s" }` as can be seen [here](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/plugins/coding.lua#L106-L123).

**Reply by sainad2222** [reply]
Yes I tried that too. Updated the description again with the configs I tried. Let me know if I am missing something else.

**Reply by dpetka2001** [reply, marked as answer]
If you run Neovim nightly then Neovim itself sets them as default keymaps when you're inside a snippet session. This is the [relevant commit](https://github.com/neovim/neovim/commit/e6cfcaed184d4ecdc8a8638429e1bd9e1b3251dc). I tested in Neovim nightly and indeed they are still there even if you disable them from LazyVim. I also tested in Neovim stable and this doesn't happen (they are disabled when you disable them from LazyVim). Maybe you should ask in Neovim github repo if there's a way to disable them.

**Reply by sainad2222** [reply]
Thank you. This might be the issue
